### PR TITLE
bug fix #10872

### LINKED ISSUE #10872: bug StochasticWeightAveraging
## 🐛 Bug

input avg_fn is not work.
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/stochastic_weight_avg.py#L204
this line self.avg_fn --> self._avg_fn

### To Reproduce

Always

### Expected behavior

[
](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/stochastic_weight_avg.py#L204
this line self.avg_fn --> self._avg_fn)
### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by qqueing**
## What does this PR do?

Fixes #10872

## Before submitting

- [v] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [v] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [v] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by rohitgr7 — COMMENTED** [review]
thanks for your contribution 😃 :
mind:
- add an entry in the changelog
- add a test that actually calls this part of this callback
?

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/callbacks/stochastic_weight_avg.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
             self.n_averaged = torch.tensor(0, dtype=torch.long, device=pl_module.device)
 
         if self.swa_start <= trainer.current_epoch <= self.swa_end:
-            self.update_parameters(self._average_model, pl_module, self.n_averaged, self.avg_fn)
+            self.update_parameters(self._average_model, pl_module, self.n_averaged, self._avg_fn)
```
Comment (Borda) [inline pytorch_lightning/callbacks/stochastic_weight_avg.py:204]: this is not tested anywhere? with this wrong ref it shall crash...

Comment (tchaton) [reply pytorch_lightning/callbacks/stochastic_weight_avg.py:204]: Let's add a unit test to prevent regression. @qqueing Mind doing so ?

**Review by tchaton — APPROVED** [review]
LGTM !
